addons/sale/models/sale_order_line.py

```diff
@@ -616,12 +616,32 @@ def _prepare_procurement_values(self, group_id=False):
         """
         return {}
 
-    @api.depends('product_id')
+    def _is_downpayment_draft(self):
+        # Verify that this SO line is
+        # first a down payment and
+        # second that the invoice(s) line are in a draft state
+        self.ensure_one()
+
+        if not self.is_downpayment:
+            return False
+
+        if any(line.move_id.state != 'draft' for line in self._get_invoice_lines()):
+            return False
+
+        return True
```

Comment:
Please make a proper docstring like the one I proposed below.

I also added a suggestion of @Demesmaeker about using `all()` and putting the result in a variable to add clarity
 
```suggestion
    def _is_downpayment_draft(self):
        """ Check that this sale order line is a draft down payment 
        
        Note: self.ensure_one()
        
        :return: True Verify that this SO line is a draft down payment 
        :rtype: Boolean
        """
        self.ensure_one()
        
        has_all_lines_in_draft = all(line.move_id.state == 'draft' for line in self._get_invoice_lines())

        return self.is_downpayment and has_all_lines_in_draft
```

Reply:
I prefer the idea of `any`, it allows for an early return and you do not have to verify each `move_id`

Reply:
I may be wrong, but I don't see your point here. Either way, it'll stop earlier if the `state != 'draft`. And for your return to be `True`, you'll have to verify each line, be it with `any` or `all`.

Reply:
No, `any` will stop whenever finding a False condition, `all` has to verify all elements in the iterator before returning, the result will be the same but depending in the number of `move_id` `any` will be faster.

Reply:
I did not read correctly the condition before, indeed you are right

Reply:
I kept the early return verifying if it is indeed a down payment before going into verifying each `move_id` line